src/targets/gpu/fuse_ops.cpp

```diff
@@ -324,6 +333,45 @@ void move_standard_front(std::vector<instruction_ref>& args)
         std::swap(*it, args.front());
 }
 
+struct find_gelu
+{
+    auto matcher() const
+    {
+        return match::name("gpu::mul")(
+            match::arg(1)(match::name("gpu::add")(match::arg(0)(match::name("gpu::erf")(match::arg(
+                0)(match::name("gpu::div", "gpu::mul")(match::arg(0)(match::any().bind("x")))))))));
```

Comment:
Why does it list both `div` and `mul`?

Reply:
The equation can be written either with a division or multiply by a reciprocal.

Reply:
But if its a multiply shouldn't `x` bind to the input of the reciprocal? It seems like you should do:

```cpp
match::any_of(
    match::name("gpu::div")(match::arg(0)(match::any().bind("x"))),
    match::name("gpu::mul")(match::arg(0)(match::name("gpu::recip")(match::arg(0)(match::any().bind("x"))))))
```

Reply:
We wouldn't have the input bound to the reciprocal in this case since it's already precomputed. It won't show up in the program after lowering.

Reply:
But that is what you are using as an input to the operator. If its precomputed then why isn't the entire operation removed by constant propagation?

Reply:
I see now. Its the second operator that is a recirpocal. Either way, you dont change the implementation if its a multiply or divide. That would make a difference. Since we transform the div to a mul, we can just match the mul instead of trying to match them both.

Reply:
I guess you assume the second operand is either sqrt(2) or reciprocal of that, but you dont actually check any of the constants in the matcher.